### PR TITLE
Disable validation of XML sitemaps and install noop entity resolver, …

### LINKED ISSUE #151: [Sitemaps] Disable XML resolvers
Sometimes processing a sitemap (including XHTML documents delivered as content when a sitemap.xml is requested) takes long (minutes), ev. causing Hadoop MapReduce tasks to fail with a timeout. These processes hang with a connection to hans-moleman.w3.org (cf. [[1](http://strangersgeospatial.blogspot.de/2013/05/hans-moleman-issues.html)])
```
java    6823 yarn  390u  IPv4             490354      0t0      TCP ip-10-143-152-145.ec2.internal:33223->hans-moleman.w3.org:http (CLOSE_WAIT)
java    6823 yarn  391u  IPv4             505696      0t0      TCP ip-10-143-152-145.ec2.internal:33228->hans-moleman.w3.org:http (ESTABLISHED)
```
opened by XMLEntityManager:
```
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1325)
        - locked <0x000000074fbc0430> (a sun.net.www.protocol.http.HttpURLConnection)
        at org.apache.xerces.impl.XMLEntityManager.setupCurrentEntity(Unknown Source)
```

To process sitemaps, there should be no need for these remote resources. The problem affects both DOM and SAX parser.

### PR DESCRIPTION
**Pull request by sebastian-nagel**
…fixes #151

### CODE REVIEW

File: src/main/java/crawlercommons/sitemaps/SiteMapParserSAX.java [hunk 1]
```diff
@@ -327,9 +329,21 @@ protected AbstractSiteMap processGzip(URL url, byte[] response) throws IOExcepti
     protected AbstractSiteMap processXml(URL sitemapUrl, InputSource is) throws UnknownFormatException {
 
         SAXParserFactory factory = SAXParserFactory.newInstance();
+        factory.setValidating(false);
+        factory.setXIncludeAware(false);
+        try {
+            factory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+        } catch (Exception e) {
+        }
```
Comment (kkrugler) [inline src/main/java/crawlercommons/sitemaps/SiteMapParserSAX.java]: Feels like we should re-throw this as a RuntimeException

File: src/main/java/crawlercommons/sitemaps/SiteMapParserSAX.java [hunk 2]
```diff
@@ -327,9 +329,21 @@ protected AbstractSiteMap processGzip(URL url, byte[] response) throws IOExcepti
     protected AbstractSiteMap processXml(URL sitemapUrl, InputSource is) throws UnknownFormatException {
 
         SAXParserFactory factory = SAXParserFactory.newInstance();
+        factory.setValidating(false);
```
Comment (kkrugler) [inline src/main/java/crawlercommons/sitemaps/SiteMapParserSAX.java]: Some comments about why we're disabling features would be useful.

File: src/main/java/crawlercommons/sitemaps/SiteMapParserSAX.java (hunk 1 above)
Comment (sebastian-nagel) [inline src/main/java/crawlercommons/sitemaps/SiteMapParserSAX.java]: Ok, should not happen anyway.

File: src/main/java/crawlercommons/sitemaps/SiteMapParserSAX.java (hunk 2 above)
Comment (sebastian-nagel) [inline src/main/java/crawlercommons/sitemaps/SiteMapParserSAX.java]: Ok.

**Comment by sebastian-nagel** [comment]
Updated PR:
- following review by @kkrugler 
- disable xinclude also for DOM parser

**Comment by sebastian-nagel** [comment]
Any objections to commit? It definitely speeds up parsing and makes any performance measures more reliable.

**Comment by jnioche** [comment]
+1 go ahead!

**Comment by kkrugler** [comment]
+1 looks good to me.

**Comment by sebastian-nagel** [comment]
Thanks!

**Pull request merged**